content/kb/tutorials/databases/mysql.md

```diff
@@ -79,32 +81,19 @@ Copy the code below to your Streamlit app and run it. Make sure to adapt `query`
 # streamlit_app.py
 
 import streamlit as st
-import mysql.connector
 
 # Initialize connection.
-# Uses st.cache_resource to only run once.
-@st.cache_resource
-def init_connection():
-    return mysql.connector.connect(**st.secrets["mysql"])
-
-conn = init_connection()
+conn = st.experimental_connection('mysql', type='sql')
 
 # Perform query.
-# Uses st.cache_data to only rerun when the query changes or after 10 min.
-@st.cache_data(ttl=600)
-def run_query(query):
-    with conn.cursor() as cur:
-        cur.execute(query)
-        return cur.fetchall()
-
-rows = run_query("SELECT * from mytable;")
+df = conn.query('SELECT * from mytable;', ttl=600)
 
 # Print results.
-for row in rows:
-    st.write(f"{row[0]} has a :{row[1]}:")
+for person, pet in zip(df['person'], df['pet']):
+    st.write(f"{person} has a :{pet}:")
```

Comment:
Pandas experts in the house - should I do this approach or something else?

Reply:
Used `df.itertuples()` a85f85a and 798e125.

Btw, in the Snowflake app code here `st.write(f"{row.person} has a :{row.pet}:")`, should PERSON and PET be capitalized? Need to verify this but IIRC Snowflake returns capitalized column names?

Reply:
snowpark-python DOES return capitals - fixed. Good catch

snowflake-sqlalchemy + SQLConnection.query() returns lower case, so I left it that way for this example.